src/rust/runtime/queue/mod.rs

```diff
@@ -54,28 +54,33 @@ impl IoQueueTable {
     /// Allocates a new entry in the target I/O queue descriptors table.
     pub fn alloc(&mut self, qtype: u32) -> QDesc {
         let index: usize = self.table.insert(qtype);
-        QDesc::from(index + Self::BASE_QD)
+
+        // Ensure that the allocation would yield to a safe conversion between usize to u32.
+        // Note: This imposes a limit on the number of open queue descriptors in u32::MAX.
+        assert!(index <= u32::MAX as usize, "I/O descriptors table overflow");
```

Comment:
Shouldn't this assert be for `index + Self::BASE_QD <= u32::MAX`?  Because after that, line 62 below could fail.

Reply:
Yest it should. It is fixed now.

Reply:
Great, thanks!